Patch to detect nvidia over nv

possible for upstream but might get rejected

--- a/hw/xfree86/common/xf86Config.c	2019-02-12 11:27:50.449786001 -0800
+++ b/hw/xfree86/common/xf86Config.c	2019-02-12 11:34:41.130603444 -0800
@@ -965,6 +965,7 @@
 freeDevice(InputInfoPtr * list, InputInfoPtr pInfo)
 {
     InputInfoPtr *devs;
+    char **end, **drv, **drivers, **x;
 
     for (devs = list; devs && *devs; devs++) {
         if (*devs == pInfo) {
@@ -974,6 +975,25 @@
             break;
         }
     }
+    /*
+     * Since the nvidia kernel driver interferes with the nv Xorg driver,
+     * make sure to prefer nvidia if it's installed
+     */
+    for (drv = drivers; drv != end; drv++) {
+       char **nv, **nvidia;
+        if (!strcmp(*drv, "nv")) {
+            nv = drv;
+            for (drv = nv; drv != end; drv++) {
+                if (!strcmp(*drv, "nvidia")) {
+                    nvidia = drv;
+                    x = *nvidia; *nvidia = *nv; *nv = x;
+                   break;
+                }
+            }
+            /* if we get here, nvidia was already ahead of nv */
+            break;
+        }
+    }
 }
 
 /**
